.github/workflows/gradle.yml

```
@@ -0,0 +1,46 @@
+name: Java CI with Gradle
+
+# opened: 新しいPull Requestが作成されたときワークフローが動く
+# synchronize: Pull Requestに対して何らかの変更(push)があったとき
+# main: mainブランチが対象
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+    branches:
+      - main
```

pushイベントの条件としてmainブランチならというoptionを入れないといけないですね。
```suggestion
  push:
    branches:
      - main
```

---

mainブランチのみを指定するのですね。
ありがとうございます。
修正します。